### ISSUE TITLE
Offline support

### ISSUE THREAD

**Issue by 573** [intercellular/cell#124]
How is the state regarding reading pages generated using celljs in tools providing offline mode i. e. pocket ?

**Comment by piranna** [comment]
So far they can be executed from file:// scheme, so pretty good :-) If you
are talking about state persistence, I think Cell don't support it (yet?)
and would need some way to re-hidrate the objects. Maybe Cell would add a
.toJson() method and another module would store and fetch the JSON-like
objects from LocalStorage...

**Comment by gliechtenstein** [comment]
I think @573 means if the dynamically generated DOM gets properly crawled by services like pocket or instapaper. I just tried and it's working fine.

But you can try yourself and let me know.

**Comment by 573** [comment]
@gliechtenstein i. e. pocket does seem to properly crawl the generated DOM, in laymans terms as long as pocket has an internet connection the celljs page is properly rendered in pocket. I tested a similar scenario with a celljs page stored in pocket and having no internet connection. That said I switched the connection state to off shortly after the page was fully rendered in pocket. As a result the page / DOM was suddenly empty again. I guess that is where @piranna suggestion comes into play...

**Comment by spyhole** [comment]
pwa +1

**Comment by gliechtenstein** [comment]
@573 Isn't this something that the developers should implement on their own? Not sure how this is related to Cell.

But I may be misunderstanding all this, so if you have a reference about this please share. Thanks!

**Comment by spyhole** [comment]
@573 give a hand with @gliechtenstein

**Comment by gliechtenstein** [comment]
Closing for now [for the reason above](https://github.com/intercellular/cell/issues/124#issuecomment-311800663) but feel free to reopen if I am misunderstanding something. Thanks!

**Comment by 573** [comment]
@gliechtenstein I issued a report to the pocket people. Thanks!
